cli/src/pcluster/templates/cluster_stack.py

```diff
@@ -1052,20 +1046,16 @@ def _add_head_node(self):
                     "cfn_fsx_options": self._get_shared_storage_options(SharedStorageType.FSX),
                     "cfn_volume": self._get_shared_storage_ids(SharedStorageType.EBS),
                     "cfn_scheduler": self._cluster_config.scheduling.scheduler,
-                    "cfn_encrypted_ephemeral": self._cluster_config.head_node.storage.ephemeral_volume.encrypted
-                    if self._cluster_config.head_node.storage
-                    and self._cluster_config.head_node.storage.ephemeral_volume
+                    "cfn_encrypted_ephemeral": head_node.storage.ephemeral_volume.encrypted
+                    if head_node.storage and head_node.storage.ephemeral_volume
                     else "NONE",
-                    "cfn_ephemeral_dir": self._cluster_config.head_node.storage.ephemeral_volume.mount_dir
-                    if self._cluster_config.head_node.storage
-                    and self._cluster_config.head_node.storage.ephemeral_volume
+                    "cfn_ephemeral_dir": head_node.storage.ephemeral_volume.mount_dir
+                    if head_node.storage and head_node.storage.ephemeral_volume
                     else "NONE",
-                    # "cfn_shared_dir": self._cluster_config.head_node.storage.root_volume.mount_dir  # TODO
-                    # if self._cluster_config.head_node.storage and self._cluster_config.head_node.storage.root_volume
+                    # "cfn_shared_dir": head_node.storage.root_volume.mount_dir  # TODO
+                    # if head_node.storage and head_node.storage.root_volume
                     # else "NONE",
-                    "cfn_proxy": self._cluster_config.head_node.networking.proxy
-                    if self._cluster_config.head_node.networking.proxy
-                    else "NONE",
+                    "cfn_proxy": head_node.networking.proxy if head_node.networking.proxy else "NONE",
```

Comment:
I am not sure "None" value works for chef client attributes json or not, did you test it? I use "" empty string instead of "NONE".

Reply:
It depends by the cookbook logic. For some of them in the recipes we have `if param != "NONE"`, I'm trying to keep the things aligned to old behavior to avoid changing the recipes for now.